src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -73,6 +79,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             {
                 // Pass through attribute that is also a well-known HTML attribute.
                 output.CopyHtmlAttribute(SrcAttributeName, context);
+
+                // Need to update the TagHelperOutput's attribute if the URL is application relative.
+                output.Attributes[SrcAttributeName].Value = Src;
```

Comment:
Doesn't play nicely with other tag helpers targeting this element and makes much of the work done in `CopyHtmlAttribute()` irrelevant. Should either be conditional e.g. done only if `output.Attributes[SrcAttributeName].Value` matches the original `Src` value or instead `UrlHelper.Content(output.Attributes[SrcAttributeName].Value)`.